### PR TITLE
feat: update light-client rpc

### PR DESCRIPTION
**Pull request by PainterPuppets**
# Description
update light-client rpc 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] test `get_cells_capacity` rpc 
- [x] test  `send_transaction` rpc
- [x] test `get_genesis_block` rpc
- [x] test `get_transaction` & `fetch_transaction` rpc
- [x] test `get_header` & `fetch_header` rpc

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by homura** [comment]
missed RPC methods

- `get_header`
- `get_transaction`
- `fetch_header`
- `fetch_transaction`

**Comment by PainterPuppets** [comment]
completed @homura

File: packages/e2e-test/tests/light-client-rpc.e2e.test.ts [hunk 1]
```diff
@@ -64,6 +68,52 @@ async function waitLightClientSync(
   }
 }
 
+async function waitLightClientFetchTransaction(
```
Comment (homura) [inline packages/e2e-test/tests/light-client-rpc.e2e.test.ts:71]: can be refactored with p-retry and p-timeout, the `await lightClientRpc.fetchTransaction` may await forever cause CI running for a long time

```ts
import pRetry from 'p-retry'
import pTimeout from 'p-timeout'

pRetry(
  () => pTimeout(
    () => {
      const result = await lightClientRpc.fetchTransaction(txHash);
      if (result.status === FetchFlag.Fetched) return result.data
      throw new Error('transaction not found')
    },
), { retries: 3, onFailedAttempt: () => asyncSleep(300) })

```

**Review by homura — APPROVED** [review]
maybe we can refactor in next PR

**Comment by gpBlockchain** [comment]
`get_transactions`'s response  is inconsistent
https://github.com/nervosnetwork/ckb-light-client#get_transactions
### ckb-indexer 
https://github.com/nervosnetwork/ckb-indexer/blob/cd5039fc2bb797c29bcd5a88a7c99345d5344c97/src/service.rs#L343-L346
```

ckb-indexer 
#[derive(Serialize)]
#[serde(untagged)]
pub enum Tx {
    Ungrouped(TxWithCell),
    Grouped(TxWithCells),
}

#[derive(Serialize)]
pub struct TxWithCell {
    tx_hash: H256,
    block_number: BlockNumber,
    tx_index: Uint32,
    io_index: Uint32,
    io_type: CellType,
}

#[derive(Serialize)]
pub struct TxWithCells {
    tx_hash: H256,
    block_number: BlockNumber,
    tx_index: Uint32,
    cells: Vec<(CellType, Uint32)>,
}

```
### ckb-light-client
https://github.com/nervosnetwork/ckb-light-client/blob/353b3c3514e74d95886fc929fbee15e17521a280/src/service.rs#L253-L256
```
pub enum Tx {
    Ungrouped(TxWithCell),
    Grouped(TxWithCells),
}

#[derive(Serialize)]
pub struct TxWithCell {
    transaction: TransactionView,
    block_number: BlockNumber,
    tx_index: Uint32,
    io_index: Uint32,
    io_type: CellType,
}

#[derive(Serialize)]
pub struct TxWithCells {
    transaction: TransactionView,
    block_number: BlockNumber,
    tx_index: Uint32,
    cells: Vec<(CellType, Uint32)>,
}

```

**Pull request merged**